packages/httpsnippet-client-api/src/index.js

```diff
@@ -242,28 +229,20 @@ module.exports = function (source, options) {
   const args = [];
 
   let accessor = method;
-  if ('operationId' in operation.schema && operation.schema.operationId.length > 0) {
-    accessor = operation.schema.operationId;
+  if (operation.hasOperationId()) {
+    accessor = operation.getOperationId({ camelCase: true });
```

Comment:
This `camelCase` option being here means that because this operation has an `operationId` we're going to do our best to try to clean it up so that it can be used as an accessor. I'm not going to recover the full logic it's doing but you can see it [here](https://github.com/readmeio/oas/blob/main/src/operation.ts#L326-L368).

What this work will allow us however to do is completely rid ourselves of the nasty array syntax that we had in place a bit below for non-alphanumeric IDs like the following:

```js
const sdk = require('api')('https://example.com/openapi.json');

sdk.auth('123');
sdk['find/pets-by-status']({status: 'available'})
  .then(res => console.log(res))
  .catch(err => console.error(err));
```

That same snippet is now much cleaner:

```js
const sdk = require('api')('https://example.com/openapi.json');

sdk.auth('123');
sdk.findPetsByStatus({status: 'available'})
  .then(res => console.log(res))
  .catch(err => console.error(err));
```

I will write tests in a followup PR within `api` to make sure that we're still able to handle backwards compatibility with the array'd operationID format though.